SE-XXX Typed throws Outline.md

```diff
@@ -0,0 +1,99 @@
+
+## General motivation
+
+???
+
+- multiple layers, public apis ???
+
+## Error scenarios considered

```

Comment:
Add those scenarios over what it is already in the main proposal under the **proposal** section

Reply:
done